app/uk/gov/hmrc/agentuserclientdetails/config/AppConfig.scala

```diff
@@ -35,8 +35,8 @@ trait AppConfig {
   val enrolmentStoreProxyUrl: String
 
   val desBaseUrl: String
-  val desEnvironment: String
-  val desAuthToken: String
+  def desEnvironment: String
```

Comment:
These should be val's?

Reply:
I think I see why you used def's - for the tests. You could mock ServicesConfig instead of AppConfig...creating a method like def mockServicesConfig(configVal: String) = mockServicesConfig.getConfString(...).expects(...).returning(configVal)

Reply:
ScalaMock can't mock vals:

"The most significant difference is that Mockito can mock vals, vars, and class members while ScalaMock can't. ScalaMock uses macros to override functions in contrast with Mockito, which uses reflection."

So without using Mockito this is, I think, the best way of getting around this. The same had been done previously with emailBaseUrl in AppConfig (line 33)